statistics/semantic/src/main/java/com/spotify/heroic/statistics/semantic/SemanticRatioGauge.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2017 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.statistics.semantic;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.RatioGauge;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class SemanticRatioGauge extends RatioGauge {
+    private final Counter numerator = new Counter();
+    private final Counter denominator = new Counter();
+
+    public SemanticRatioGauge(final long numerator, final long denominator) {
+        this.numerator.inc(numerator);
+        this.denominator.inc(denominator);
+    }
+
+    public void setNumerator(final long value) {
+        numerator.dec(numerator.getCount());
+        numerator.inc(value);
+    }
+
+    public void incNumerator() {
+        numerator.inc();
+    }
+
+    public void decNumerator() {
+        numerator.dec();
+    }
+
+    public void setDenominator(final long value) {
+        denominator.dec(denominator.getCount());
+        denominator.inc(value);
```

Comment:
Awesome, yes, it would feel much better. I chose Counter since it was mentioned earlier. I'll fix it :)